server/internal/client/ollama/registry.go

```diff
@@ -490,77 +481,61 @@ func (r *Registry) Pull(ctx context.Context, name string) error {
 			t.update(l, 0, err)
 			continue
 		}
+		// TODO(bmizerany): fix this unbounded use of defer
 		defer chunked.Close()
 
 		var progress atomic.Int64
 		for cs, err := range r.chunksums(ctx, name, l) {
 			if err != nil {
+				// Bad chunksums response, update tracing
+				// clients and then bail.
 				t.update(l, progress.Load(), err)
-				break
+				return err
```

Comment:
It looks like this will abort the full download in the case of one chunk checksum failing, is that accurate? If that is the case maybe some retry logic would make sense (could be a follow-on change).

Reply:
That is correct. This had retry/backoff in a previous iteration, but I removed it in favor of clients of the package retrying instead. Back-pressure is a wonderful drug.

However, in the current form, it will abort all in progress chunk downloads that are legit, so if the chunksums response is cutoff, we should allow the current chunks to keep going. We should also still update trace right away too. Then the clients can determine if they want to abort all-together or not but canceling the context that passed to Pull.

Reply:
I'll make this change in a follow-up PR.